contracts/router/RouterV2.sol

```diff
@@ -0,0 +1,175 @@
+    mapping(bytes32 => address) public idToModule;
+
+    /// @inheritdoc IRouterV2
+    mapping(address => bytes32) public moduleToId;
+
+    event ModuleConnected(bytes32 moduleId, address indexed bridgeModule);
+
+    constructor(address _swapQuoter) {
+        swapQuoter = ISwapQuoterV1(_swapQuoter);
+    }
+
+    /// @inheritdoc IRouterV2
+    function bridgeViaSynapse(
+        address to,
+        uint256 chainId,
+        bytes32 moduleId,
+        address token,
+        uint256 amount,
+        SwapQuery memory originQuery,
+        SwapQuery memory destQuery
+    ) external payable {
+        address bridgeModule = idToModule[moduleId];
+        if (bridgeModule == address(0)) revert ModuleNotExists();
+
+        // pull (and possibly swap) token into router
+        if (_hasAdapter(originQuery)) {
+            (token, amount) = _doSwap(address(this), token, amount, originQuery);
+        } else {
+            _pullToken(address(this), token, amount);
+        }
+
+        // delegate bridge call to module
+        // @dev delegatecall should approve to spend
+        bytes memory payload = abi.encodeWithSelector(
+            IBridgeModule.delegateBridge.selector,
+            to,
+            chainId,
+            token,
+            amount,
+            destQuery
+        );
+        (bool success, ) = bridgeModule.delegatecall(payload);
+        if (!success) revert BridgeFailed();
+    }
+
+    /// @inheritdoc IRouterV2
+    function swap(
+        address to,
+        address token,
+        uint256 amount,
+        SwapQuery memory query
+    ) external payable returns (uint256 amountOut) {
+        if (!_hasAdapter(query)) revert QueryEmpty();
+
+        address tokenOut;
+        (tokenOut, amountOut) = _doSwap(to, token, amount, query);
+    }
+
+    /// @inheritdoc IRouterV2
+    function connectBridgeModule(bytes32 moduleId, address bridgeModule) external onlyOwner {
+        if (idToModule[moduleId] != address(0)) revert ModuleExists();
+
+        _bridgeModules.push(bridgeModule);
+        idToModule[moduleId] = bridgeModule;
+        moduleToId[bridgeModule] = moduleId;
+
+        emit ModuleConnected(moduleId, bridgeModule);
+    }
+
+    /// @inheritdoc IRouterV2
+    function getDestinationBridgeTokens(address tokenOut) external view returns (BridgeToken[] memory destTokens) {
+        BridgeToken[][] memory unflattenedDestTokens = new BridgeToken[][](_bridgeModules.length);
+        uint256 destTokensLength;
+
+        for (uint256 i = 0; i < _bridgeModules.length; ++i) {
+            BridgeToken[] memory bridgeTokens = IBridgeModule(_bridgeModules[i]).getBridgeTokens();
+
+            // assemble limited token format for quoter call
+            LimitedToken[] memory bridgeTokensIn = new LimitedToken[](bridgeTokens.length);
+            for (uint256 j = 0; j < bridgeTokens.length; ++j) {
+                bridgeTokensIn[j] = LimitedToken({actionMask: uint256(Action.Swap), token: bridgeTokens[j].token});
```

Comment:
`actionMask` here depends on the "type" of the bridge token.

It should be queried from the bridge module using a not-yet-present-in-the-interface getter :)

Reply:
https://github.com/synapsecns/synapse-contracts/pull/273 should do it for the getter